### DISCUSSION TITLE
Does py_canoe support flow control of CAN bus messages?

### DISCUSSION THREAD

**Discussion by jromehenry** [chaitu-ycr/py_canoe#57] [Q&A]
Does py_canoe support flow control of CAN bus messages? 
If yes, a) through which API methods? b) what paramaters can I specify in my tx request?
I'm attempting to do the following: x.send_diag_request(diag_ecu_qualifier_name = "EIM", request = "22 FE 00").
This python code from tester reads DID 0xFE00. Can the tester also provide any flow control info in this request to the product? Please expand on how py_canoe's flow control works? thanks kindly

**Comment by chaitu-ycr** [comment]
flow control will be taken care by CANoe application.
py_canoe will receive full data from CANoe when requested to read data from ECU.
when writing you need to send all bytes from py_canoe.
py_canoe doesn't have restrictions if you are sending diagnostic request more than 8 bytes.

 with py_canoe it is not possible to control/change data flow between ECU and CANoe.
